Propose Build Interest Group Governance Charter (#31)
opensearch-project/technical-steering: special-interest-groups/build-interest-group-charter.md

# OpenSearch Project Build Interest Group (BIG) Governance Charter

## Purpose

> The OpenSearch Project, following Technical Steering Committee (TSC) recommendations, proposes to create a Build Interest Group (BIG) that focuses on repository health. This includes managing new repository creation, monitoring activity in existing repositories, and tracking maintainer status. This group will also establish guidelines for members, set criteria for newly proposed repositories, and come up with recommendations which will be then reviewed by TSC for decision making. The group will provide feedback to the Technical Steering Committee (TSC) before the final decision is made through a voting of TSC members.

Reviewer: I think BUG should also take care of archiving (or retiring) the existing repositories, @peterzhuamazon wdyt? Something along these lines:

```suggestion
The OpenSearch Project, following Technical Steering Committee (TSC) recommendations, proposes to create a Build Interest Group (BIG) that focuses on repository health. This includes managing new repository creation, retiring or archiving the existing repositories, monitoring activity in existing repositories, and tracking maintainer status. This group will also establish guidelines for members, set criteria for newly proposed repositories, and come up with recommendations which will be then reviewed by TSC for decision making. The group will provide feedback to the Technical Steering Committee (TSC) before the final decision is made through a voting of TSC members.
```

Author: LGTM, thanks!

> The OpenSearch Project, following Technical Steering Committee (TSC) recommendations, proposes to create a Build Interest Group (BIG) that focuses on repository health. This includes managing new repository creation, monitoring activity in existing repositories, and tracking maintainer status. This group will also establish guidelines for members, set criteria for newly proposed repositories, and come up with recommendations which will be then reviewed by TSC for decision making. The group will provide feedback to the Technical Steering Committee (TSC) before the final decision is made through a voting of TSC members.

Reviewer 2: ```suggestion
The OpenSearch Project, following Technical Steering Committee (TSC) recommendations, proposes to create a Build Interest Group (BIG) that focuses on repository health. This includes managing new repository creation, monitoring activity in existing repositories, and tracking maintainer status. This group will also establish guidelines for members, set criteria for newly proposed repositories, and come up with recommendations which will be then reviewed by TSC for decision making. The group will provide feedback to the TSC before the final decision is made through a voting of TSC members.
```

Author: Update in different commit now, thanks.

## Responsibilities

> * Start an initial review of the new repository's proposal / RFC / Issue (through a [new repository request](https://github.com/opensearch-project/.github/issues/new?template=REPOSITORY_REQUEST_TEMPLATE.yaml) send by the requester)

Reviewer 2: ```suggestion
* Initiate a review of the new repository's proposal / RFC / Issue through a [new repository request](https://github.com/opensearch-project/.github/issues/new?template=REPOSITORY_REQUEST_TEMPLATE.yaml)
```

  * Based on [these categories](https://github.com/opensearch-project/.github/issues/296)
* Ensure new repositories align with the foundation’s [charter](https://foundation.opensearch.org/assets/media/OpenSearch%20Project%20Technical%20Charter%20Final%209-13-2024.docx.pdf)
* Review by a subject matter expert (SME) on the proposal

>   * For example, BIG will help contact maintainers from ML Commons repository to review machine learning, neural search, and k-NN related proposals

Reviewer 2: ```suggestion
  * For example, BIG will help contact maintainers from ml-commons repository to review machine learning, neural search, and k-NN related proposals
```

> * Review the naming, proposal topic, and code (if available) in details

Reviewer 2: ```suggestion
* Review the naming, proposal topic, and code (if available) in detail
```

> * If the repository needs to be included in a future OpenSearch Release bundle, ensure a SME from the OpenSearch Release Team reviews the timelines

Reviewer 2: ```suggestion
* If the repository needs to be included in a future OpenSearch release bundle, ensure a SME from the OpenSearch Release Team reviews the timelines
```
